File: algorithms/1_easy/H/03602_hexadecimal_and_hexatrigesimal_conversion/solution.cpp
class Solution {
    string to_custom_base(int num, const int base) {
        if (num == 0) {
            return "0";
        }

        string res{};

        while (num > 0) {
            const int remainder{num % base};
            if (remainder < 10) {
                res.push_back(static_cast<char>(remainder + '0'));
            } else {
                res.push_back(static_cast<char>(remainder - 10 + 'A'));
            }

            num /= base;
        }

        reverse(res.begin(), res.end());

        return res;
    }
public:
    string concatHex36(int n) {
        return to_custom_base(n * n, 16) + to_custom_base(n * n * n, 36);
    }
};

File: algorithms/1_easy/H/03602_hexadecimal_and_hexatrigesimal_conversion/solution.py
class Solution:
    def to_custom_base(self, num: int, base: int) -> str:
        if num == 0:
            return "0"

        res = []

        while num > 0:
            remainder = num % base
            if remainder < 10:
                res.append(chr(remainder + ord('0')))
            else:
                res.append(chr(remainder - 10 + ord('A')))

            num //= base

        res.reverse()

        return "".join(res)

    def concatHex36(self, n: int) -> str:
        return "{}{}".format(
            self.to_custom_base(n * n, 16),
            self.to_custom_base(n * n * n, 36)
        )
